File: BE-FindingJob/src/main/java/com/example/befindingjob/model/Application.java
package com.example.befindingjob.model;

import com.example.befindingjob.model.enumModel.ApplicationStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "applications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Application {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer applicationId;

    @ManyToOne
    @JoinColumn(name = "job_id", nullable = false)
    private Job job;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private java.time.LocalDateTime applicationDate;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    @Column(columnDefinition = "TEXT")
    private String coverLetter;

    @OneToOne(mappedBy = "application", cascade = CascadeType.ALL, orphanRemoval = true)
    private Resume resume;
}

File: JobHub/app/src/main/java/com/example/jobhub/entity/Application.kt
package com.example.jobhub.entity

import android.os.Parcelable
import com.example.jobhub.entity.enumm.ApplicationStatus
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime

@Parcelize
data class Application(
    val applicationId: Int,
    val job: Job,
    val user: User,
    val applicationDate: LocalDateTime?,
    val status: ApplicationStatus?,
    val coverLetter: String?,
    val resume: Resume?
) : Parcelable
